src/providers/Bee.tsx

```diff
@@ -61,6 +61,14 @@ interface ContextInterface {
   refresh: () => Promise<void>
 }
 
+const urlSearchParams = new URLSearchParams(window.location.search)
+const newApiKey = urlSearchParams.get('v')
+
+if (newApiKey) {
+  localStorage.setItem('apiKey', newApiKey)
+  window.location.search = ''
+}
+
```

Comment:
Could you extract this into `SettingsProvider` and store there? That is the top level provider and IMHO makes sense to also store the key in info. It already passes down the `Bee` object.